src/Printers/Models/PrinterModel.ts

```diff
@@ -0,0 +1,227 @@
+    // E or 8  =  203.2 mm/sec. (8 inches/sec.)
+    // 9       =  220.5 mm/sec. (9 inches/sec.)
+    // 10      =  245 mm/sec.   (10 inches/sec.)
+    // 11      =  269.5 mm/sec. (11 inches/sec.)
+    // 12      =  304.8 mm/sec. (12 inches/sec.)
+    // 13      =  13 in/sec
+    // 14      =  14 in/sec
+    // This gets encoded into the speed table.
+    // Every speed table should also have entries for ipsPrinterMin, ipsPrinterMax, and auto.
+    // These should be duplicate entries of real values in the speed table so that
+    // we have sane defaults for commands to default to.
+    abstract get speedTable(): ReadonlyMap<PrintSpeed, number>;
+    abstract get maxDarkness(): number;
+
+    public isSpeedValid(speed: PrintSpeed): boolean {
+        return this.speedTable.has(speed);
+    }
+
+    public getSpeedValue(speed: PrintSpeed): number | undefined {
+        const val = this.speedTable.get(speed) ?? this.speedTable[PrintSpeed.auto];
+        console.log('Found speed ', val, 'for speed named', PrintSpeed[speed]);
+        return val;
+    }
+
+    public fromRawSpeed(rawSpeed: number): PrintSpeed {
+        for (const [key, val] of this.speedTable) {
+            if (
+                val === rawSpeed &&
+                key != PrintSpeed.auto &&
+                key != PrintSpeed.ipsPrinterMax &&
+                key != PrintSpeed.ipsPrinterMin
+            ) {
+                return key;
+            }
+        }
+        return PrintSpeed.auto;
+    }
+}
+
+export class UnknownPrinter extends BasicPrinterInfo {
+    get commandLanguage(): PrinterCommandLanguage {
+        return PrinterCommandLanguage.none;
+    }
+    get speedTable(): ReadonlyMap<PrintSpeed, number> {
+        return null;
+    }
+    get model(): PrinterModel {
+        return PrinterModel.unknown;
+    }
+    get dpi(): number {
+        throw new WebZlpError('Unknown printer, cannot read metadata.');
+    }
+    get maxDarkness(): number {
+        throw new WebZlpError('Unknown printer, cannot read metadata.');
+    }
+}
+
+/** EPL printers have a lot in common. */
+export abstract class EplPrinter extends BasicPrinterInfo {
+    get commandLanguage(): PrinterCommandLanguage {
+        return PrinterCommandLanguage.epl;
+    }
+
+    get maxDarkness(): number {
+        return 15; // EPL max darkness
+    }
+}
+
+/** 28XX model printers are mostly the same */
+export abstract class LP28XX extends EplPrinter {
+    get dpi(): number {
+        return 203;
+    }
+
+    abstract get model(): PrinterModel;
+
+    private _speedTable = new Map<PrintSpeed, number>([
+        [PrintSpeed.auto, 3],
+        [PrintSpeed.ipsPrinterMax, 4],
+        [PrintSpeed.ipsPrinterMin, 1],
+        [PrintSpeed.ips1_5, 1],
+        [PrintSpeed.ips2, 2],
+        [PrintSpeed.ips2_5, 3],
+        [PrintSpeed.ips3_5, 4]
+    ]);
+    get speedTable(): ReadonlyMap<PrintSpeed, number> {
+        return this._speedTable;
+    }
+}
+
+export class LP2844 extends LP28XX {
```

Comment:
Would put these into their own files so that User's know to add new printers they just need to add new files rather than having to edit this one.

Reply:
Added a TODO to add explicit extensibility points for defining your own printers. I've learned recently just the sheer number of different printers from _other manufacturers_ that implemented ZPL emulation. Other things like the PrinterUSBManager class are going to also need extension points to support these, creating a full enumeration of them would take a lot of research (and ebay purchases).